java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/debugging/Debugger.java

```diff
@@ -48,4 +60,41 @@ public static void startDebugger(InputStream in, OutputStream out) {
             Exceptions.printStackTrace(ex);
         }
     }
+    
+    private static class ConsumeWithLookup implements MessageConsumer {
+        private final MessageConsumer delegate;
+        private OperationContext topContext;
+
+        public ConsumeWithLookup(MessageConsumer delegate) {
+            this.delegate = delegate;
+        }
+        
+        @Override
+        public void consume(Message message) throws MessageIssueException, JsonRpcException {
+            InstanceContent ic = new InstanceContent();
+            ProxyLookup ll = new ProxyLookup(new AbstractLookup(ic), Lookup.getDefault());
+            // HACK: piggyback on LSP's client.
+            if (topContext == null) {
```

Comment:
OK, probably enough for 12.2.1. Would be good to have this recorded in an issue tracker.

Reply:
Tracked as [NETBEANS-5218](https://issues.apache.org/jira/browse/NETBEANS-5218)